File: integrations/copilot-sdk/typescript/examples/agentic_chat_reasoning.ts
import type { CopilotClientPort } from "../dist/index.js";
import { defineAgent } from "./base.js";

/** Reasoning deltas from the runtime stream as AG-UI REASONING_* events. */
export const createAgenticChatReasoningAgent = (client: CopilotClientPort) =>
  defineAgent(client, {
    agentId: "agentic_chat_reasoning",
    description: "Agentic chat that surfaces the model's reasoning",
    instructions: "You are a helpful assistant. Think carefully before you answer.",
    sessionConfig: { reasoningEffort: "medium" },
  });

File: integrations/copilot-sdk/python/examples/agents/agentic_chat_reasoning.py
"""Reasoning deltas from the runtime stream as AG-UI REASONING_* events."""

from typing import Any

from ag_ui_copilot_sdk import CopilotAgent

from .base import define_agent


def create_agentic_chat_reasoning_agent(client: Any) -> CopilotAgent:
    return define_agent(
        client,
        name="agentic_chat_reasoning",
        instructions="You are a helpful assistant. Think carefully before you answer.",
        session_options={"reasoning_effort": "medium"},
    )
